### ISSUE TITLE
'make gettext' fails when the '.. todolist::' directive is present.

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1364]
There's some kind of incompatibility/regression in 1.2.1 with the 'todo' extension and the internationalisation functionality. I get the following stack trace when attempting to build the .po files when the output generator hits a '.. todolist::' directive. This bug doesn't seem to affect 1.1.3.

```
#!python

vagrant@precise64:/vagrant/fresh$ cat /tmp/sphinx-err-PyS6A7.log 
# Sphinx version: 1.2.1
# Python version: 2.7.3
# Docutils version: 0.11 release
# Jinja2 version: 2.7.2
# Loaded extensions:
#   sphinx.ext.todo from /vagrant/env/local/lib/python2.7/site-packages/sphinx/ext/todo.pyc
#   sphinx.ext.oldcmarkup from /vagrant/env/local/lib/python2.7/site-packages/sphinx/ext/oldcmarkup.pyc
Traceback (most recent call last):
  File "/vagrant/env/local/lib/python2.7/site-packages/sphinx/cmdline.py", line 254, in main
    app.build(force_all, filenames)
  File "/vagrant/env/local/lib/python2.7/site-packages/sphinx/application.py", line 212, in build
    self.builder.build_update()
  File "/vagrant/env/local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 214, in build_update
    'out of date' &#37; len(to_build))
  File "/vagrant/env/local/lib/python2.7/site-packages/sphinx/builders/gettext.py", line 174, in build
    I18nBuilder.build(self, docnames, summary, method)
  File "/vagrant/env/local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 276, in build
    self.write(docnames, list(updated_docnames), method)
  File "/vagrant/env/local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 320, in write
    self._write_serial(sorted(docnames), warnings)
  File "/vagrant/env/local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 333, in _write_serial
    self.write_doc(docname, doctree)
  File "/vagrant/env/local/lib/python2.7/site-packages/sphinx/builders/gettext.py", line 101, in write_doc
    catalog.add(msg, node)
  File "/vagrant/env/local/lib/python2.7/site-packages/sphinx/builders/gettext.py", line 62, in add
    self.metadata[msg].append((origin.source, origin.line, origin.uid))
AttributeError: 'title' object has no attribute 'uid'

```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1364
- Originally reported by: [tveastman](https://bitbucket.org/tveastman)
- Originally created at: 2014-01-28T05:37:19.618

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-01-28 05:05:53+00:00_

Thanks reporting!
I'll check in this week.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-02-28 01:57:06+00:00_

Sorry I have no time last 1 month.. I'll check ASAP.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-03-01 04:39:52+00:00_

Confirmed the behavior.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-03-02 05:42:52+00:00_

Fix 'make gettext' fails when the '.. todolist::' directive is present. Closes #1364

Nodes that are replicated like todo don't have a uid, however i18n also unnecessary.
If extensions need to choose i18n for the extended nodes or not, we need another approach.

→ <<cset 194e50fd2d0fee64c63a5374746c28edafb40444>>
